### ISSUE TITLE
serverset_sd_config problem

### ISSUE THREAD

**Issue by z-haohao** [VictoriaMetrics/VictoriaMetrics#4746]
### Is your question request related to a specific component?

Whether prometheus serverset_sd_config mechanisms will be compatible in the future, or other service discovery mechanisms can meet the above requirements.

### Describe the question in detail

When I migrated the indicators on the prometheus to the vm, I found that the service discover of the vm is not compatible with the prometheus serverset_sd_config, so whether it will be compatible in the future or there are other ways
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/125655392/0aada7b5-f971-4da6-9fd2-ac8b5e557129)

### Troubleshooting docs

- [ ] General - https://docs.victoriametrics.com/Troubleshooting.html
- [ ] vmagent - https://docs.victoriametrics.com/vmagent.html#troubleshooting
- [ ] vmalert - https://docs.victoriametrics.com/vmalert.html#troubleshooting

**Comment by Amper** [comment]
Hey @z-haohao.
This feature (`serverset_sd_config`) is not supported at this time, I've reclassified your ticket as a feature-request.
We will discuss the possibility of including this feature into our plan. And PRs is always welcome )

**Comment by hagen1778** [comment]
@z-haohao as @Amper mentioned, VM doesn't support this type of service discovery yet. You can find the full list of supported SDs here: https://docs.victoriametrics.com/sd_configs.html

**Comment by sh-cho** [comment]
I'd like to work on this

**Comment by hagen1778** [comment]
@sh-cho thanks! Assigned it to you!

**Comment by sh-cho** [comment]
![zookeeper serverset drawio (1)](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/11611397/de6c9ba4-ac56-41ca-8e04-6a68b232a26e)

@z-haohao 
Could you check if I understand serverset sd correctly? 

```
# Paths can point to a single serverset, or the root of a tree of serversets.
```
According to this comment in the prometheus [serverset_sd_config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#serverset_sd_config) documentation,

- If I add path `/svc1` (root of serversets), scrape targets are all grand-children (`.../node_001`, `.../node_002`, `.../node_1234`)
- If I add path `/svc1/app1` (single serverset), scrape targets are its children (`.../node_001`, `.../node_002`)

FYI, It may take some time to implement, since I'm not very used to golang 😅

Maybe using some [http_sd](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#http_sd_config) proxy for serverset_sd is useful in the meantime.

**Comment by z-haohao** [comment]
thank you   very  much .@sh-cho      i  have  use  it  by  this  airticle .            https://my.oschina.net/u/4447604/blog/10086185      
 and my zk show 
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/125655392/1f8f7b0e-bb27-4633-a292-8a54fd63da4c)
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/125655392/d9bb53dd-0c65-43d8-af69-48713a7115d8)
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/125655392/5d9b53bd-21ce-4d4a-bbe1-12b2b6bbe659)
It is achieved in this way.

**Comment by z-haohao** [comment]
@sh-cho Yes, you understand it correctly, which is achieved through your flowchart. Don't worry about time. I'm also trying to learn the go language and try to understand the code. If you can, I hope you can achieve it. Thank you very much.

**Comment by lf2180** [comment]
serverset_sd_config Has this discovery mechanism been developed? We need this discovery

**Comment by sh-cho** [comment]
@lf2180 Sorry I forgot this 😞 
It will be done soon (maybe in a month)

**Comment by lf2180** [comment]
Thank you very much. We need this feature.
